samples: mesh: nrf52: removed bugs in Gen. level Move GET & Publish

vikrant8052 opened the pull request:
Due to recent changes some bugs had introduced in Gen. Level Move
GET & Publish functionalities. Now separatly introduced
gen_level_move_get() and gen_level_move_publisher() to resolved it.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: removed bug in Gen. level Move GET & Publish

Due to recent changes some bugs had introduced in Gen. Level Move
GET & Publish functionalities. Now separatly introduced
gen_level_move_get() and gen_level_move_publisher() to resolved it.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +648,63 @@ static void gen_delta_set(struct bt_mesh_model *model,
+	}
+}
+
+static void gen_level_move_publisher(struct bt_mesh_model *model)
```
I know this is not the only function with the issue, but function names should have a verb as the last component. Here you have a noun. I.e. the name should be gen_level_move_publish()

vikrant8051 replied:
Done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +648,63 @@ static void gen_delta_set(struct bt_mesh_model *model,
+	struct net_buf_simple *msg = model->pub->msg;
+	struct generic_level_state *state = model->user_data;
+
+	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
```
```
if (model->pub->addr == BT_MESH_ADDR_UNASSIGNED) {
         ...
}
```

Helps you eliminate unnecessary indentation, which in turn eliminates unnecessary breaks of long lines (I know this is not an issue for this particular function, but it's good to get into the habit and be consistent with this kind of function flow).

jhedberg replied:
(edited above comment - "..." was supposed to be "return")

vikrant8051 replied:
Done !!

jhedberg reviewed (changes requested):
A few minor things

vikrant8052 committed:
samples: mesh: nrf52: rename some functions name

Function names should have a verb as the last component.
As per this standard, rename some fuunctions names.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: improved coding style

Improved coding style.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
+		}
+
+		/* This is as per PTS requirement */
+		net_buf_simple_add_u8(msg, 0x3F);
```
You might want to have a define for 0x3F instead of having it as a magic constant here. As per section 3.1.3.1 in the Mesh Model spec it means "The value is unknown". Saying that it's a PTS requirement isn't helpful since the PTS only follows the Test Specification which in turn follows the Model Specification.

vikrant8051 replied:
Done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
+			net_buf_simple_add_le16(msg, INT16_MIN);
+		} else if (state->last_delta > 0) {
+			net_buf_simple_add_le16(msg, INT16_MAX);
+		}
```
What if last_delta is 0? I that case you'd create a corrupted message without a Target Level field.

vikrant8051 replied:
when client sends delta = 0 means stop the transition.
Plus it will also set state->transition->counter = 0

That means following lines would not execute in that conditions

```
if (state->transition->counter) {
		if (state->last_delta < 0) {
			net_buf_simple_add_le16(msg, INT16_MIN);
		} else if (state->last_delta > 0) {
			net_buf_simple_add_le16(msg, INT16_MAX);
		}

		/* This is as per PTS requirement */
		net_buf_simple_add_u8(msg, UNKNOWN_VALUE);
}

```

vikrant8051 replied:
Done !!

vikrant8051 replied:
done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
+	net_buf_simple_add_le16(msg, state->level);
+
+	if (state->transition->counter) {
+
```
minor: redundant empty line

vikrant8051 replied:
done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
+	net_buf_simple_add_le16(msg, state->level);
+
+	if (state->transition->counter) {
+
```
minor: redundant empty line

vikrant8051 replied:
done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
+			net_buf_simple_add_le16(msg, INT16_MIN);
+		} else if (state->last_delta > 0) {
+			net_buf_simple_add_le16(msg, INT16_MAX);
+		}
```
See my other related comment

vikrant8051 replied:
As mentioned earlier, these lines would not get executed when
delta = 0 because when it is zero  `state->transition->counter` become zero.

vikrant8051 replied:
Done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -679,6 +650,64 @@ static void gen_delta_set(struct bt_mesh_model *model,
+		}
+
+		/* This is as per PTS requirement */
+		net_buf_simple_add_u8(msg, 0x3F);
```
See my other related comment.

vikrant8051 replied:
done !!

jhedberg commented on samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c:
```diff
@@ -961,21 +991,22 @@ static void gen_onpowerup_status(struct bt_mesh_model *model,
 
-		bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_2(0x82, 0x12));
-		net_buf_simple_add_u8(msg, state->onpowerup);
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_2(0x82, 0x12));
```
Why do you have these magic 0x82, 0x12 values here when in other places you have proper defines for the opcodes (e.g. `BT_MESH_MODEL_OP_GEN_LEVEL_STATUS`). Please use proper descriptive defines instead of magic values. Otherwise you're just making it hard to follow the code and understand exactly which message is being sent.

vikrant8051 replied:
I will take care of this in separate PR.

vikrant8051 replied:
Done !!

vikrant8052 committed:
samples: mesh: nrf52: removed bugs in Target values calculation

Target values should change as per min. & max. values of
bound states. This commit handles this.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: state binding improvements

Corrected state binding. With this commit Light Lightness
actual state would not cross its upper & lower limit.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

vikrant8052 committed:
samples: mesh: nrf52: improved code readability

Improved code readability by defining some macros.

Signed-off-by: Vikrant More <vikrant8051@gmail.com>

jhedberg commented:
recheck

vikrant8051 commented:
All suggested changes has done.
